### PR TITLE
[OpenAI] Modes required to be public in Java SDK

### PR DESCRIPTION
**Pull request by mssfang**
These classes are already in the public `/model` folder, we should keep them out in Java at least.

### CODE REVIEW

**Review by trrwilson — APPROVED** [review]

File: specification/cognitiveservices/OpenAI.Inference/client.tsp [hunk 1]
```diff
@@ -19,3 +19,27 @@ using Azure.ClientGenerator.Core;
...
+@@access(Azure.OpenAI.AudioTranslation, Access.public, "java");
+@@access(Azure.OpenAI.AudioTranslationSegment, Access.public, "java");
+@@access(Azure.OpenAI.AzureChatExtensionConfiguration, Access.public, "java");
+@@access(Azure.OpenAI.AzureChatExtensionType, Access.public, "java");
+@@access(Azure.OpenAI.AzureOpenAIOperationState, Access.public, "java");
+@@access(Azure.OpenAI.ChatChoice, Access.public, "java");
+@@access(Azure.OpenAI.ChatCompletions, Access.public, "java");
+
+// Error: Model 'ChatCompletionsOptions' cannot be set as access=public, as it refers Union 'FunctionCallPreset | FunctionName'
+// @@access(Azure.OpenAI.ChatCompletionsOptions, Access.public, "java");
```
Comment (mssfang) [inline specification/cognitiveservices/OpenAI.Inference/client.tsp:36]: @weidongxu-microsoft 

We have customized this `ChatCompletionsOptions` class, which hides some generated methods, and introduced some new implementations to it. 

 @Generated
    public FunctionCallModelBase getFunctionCall() {
        return this.functionCall;
    }

should replaced by 
  public FunctionCallConfig getFunctionCall() {
        return this.functionCallConfig;
    }

**Review by weidongxu-microsoft — APPROVED** [review]
I will see if we can release the "Union as BinaryData" typespec-java next week.

Code is merged, pending for me to test it on the OpenAI tsp and configure (safer to do this before release of emitter).

**Review by jpalvarezl — APPROVED** [review]

**Comment by weidongxu-microsoft** [comment]
I am updating typespec-java to 0.11.0 (which include Union as BinaryData) https://github.com/Azure/azure-sdk-for-java/pull/37629

I will try to see if I need some change to client.tsp, built upon this PR.

**Comment by mssfang** [comment]
@lmazuel Can I merge this PR? @joseharriaga opened a [PR ](https://github.com/Azure/azure-sdk-for-net/pull/39909) in .NET side to fix the failure in .net.

**Pull request merged**